src/codechicken/nei/recipe/GuiRecipe.java

```diff
@@ -94,6 +94,42 @@ protected GuiRecipe(GuiScreen prevgui) {
         
     }
 
+    /**
+     * Many old mods assumed a fixed NEI window height of {@code 166} pixels. Now that this is no
+     * longer the case, their fluid mouseover handling is broken. This helper class fixes these old
+     * mods by hacking the {@link #height} property's value so that these old mods' calculations
+     * return the correct value for the new height.
+     *
+     * <p>This class is an {@link AutoCloseable} so that it can be used with try-with-resources,
+     * which will ensure that {@link #height} is returned to the correct value afterwards.
+     */
+    private class HeightHack implements AutoCloseable {
+        private final int trueHeight;
+
+        private HeightHack() {
+            trueHeight = height;
+
+            if (NEIClientConfig.heightHackHandlers.contains(handler.getHandlerId())) {
+                // The old mods use the calculation ((height - 166) / 2) to compute the y-value of
+                // the top edge of the NEI window.
+                // guiTop is exactly that: the y-value of the top edge of the NEI window.
+                // So we set height to be equal to the inverse of this calculation, applied to
+                // guiTop, so that old mods get the right result back when they use this calculation.
+                height = (2 * guiTop) + 166;
+
+                // For reference, in case we ever need to modify width as well:
+                // the legacy calculation used for width is ((width - 176) / 2), which should
+                // evaluate to be equal to guiWidth (the x-value of the left edge of the NEI window).
+                // So if we wanted to override width as well, we'd do this:
+                // width = (2 * guiWidth) + 176;
+            }
+        }
+
+        @Override
+        public void close() {
+            height = trueHeight;
+        }
+    }
```

Comment:
You could also make a handler method for this, like:

```java
private void ensuringCorrectHeight(Runnable action) {
   if(NEIClientConfig.heightHackHandlers.contains(handler.getHandlerId())) {
       int originalHeight = height;
       height = (2 * guiTop) + 166;
       action.invoke();
       height = originalHeight;
   } else {
       action.invoke()
   }
}
```

and use it later like:

```java
 ensuringCorrectHeight( () ->  {
      for (int recipe : getRecipeIndices())
          if (handler.keyTyped(this, c, i, recipe))
              return;
 })
```

this way nobody will make a mistake of forgetting to put it in try-with-resources. What do you think?